spec/services/export/report_spec.rb

```diff
@@ -286,3 +288,102 @@ def link_for_row(row)
     row[@headers_for_report.length + 7]
   end
 end
+
+# This entire spec is for when there aren't forecasts but there are activities -- it's essentially
+# a copy of the setup for the above but we never add a value to the forecast.
+RSpec.describe Export::Report do
```

Comment:
I think this would be better served in a `context "there are activities with no forecasts"` block, rather than duplicating the setup for this pretty complex test suite (which I'd love us to cut down in future).

Inspired by Cristina's comment, I've checked this out and run it locally and it looks like everything is passing without us setting the following variables:
- `@implementing_organisation`
- `@third_party_project`
- `previous_report`
- `@forecast`
- `@comment`

Is it worth us adding a new top level `@report_with_activities_but_no_forecasts` (or something) for use in this new context so we don't need to create and persist a lot more entities in the database for this test? The test suite is painfully slow at the moment as-is, so I'm reluctant to introduce more persistence!

Reply:
so dragon and i tried a number of ways to avoid the duplicated set-up when we wrote this, but had no luck. i'll have another poke around today and see if i can get it to work another way. agree about the vars that don't need setting though